AEMG/training.py

```diff
@@ -87,19 +89,20 @@ def forward(self, x_t, x_tau):
 
         return (x_t, x_tau, x_t_pred, z_tau, z_tau_pred, x_tau_pred_dyn)
 
-    def dynamics_losses(self, foward_pass, weight):
-        x_t, x_tau, x_t_pred, z_tau, z_tau_pred, x_tau_pred_dyn = foward_pass
+    def dynamics_losses(self, forward_pass, weight):
+        x_t, x_tau, x_t_pred, z_tau, z_tau_pred, x_tau_pred_dyn = forward_pass
 
-        loss_ae1 = self.criterion(x_t, x_t_pred)
-        loss_ae2 = self.criterion(x_tau, x_tau_pred_dyn)
-        loss_dyn = self.criterion(z_tau_pred, z_tau)
+        loss_ae1 = self.dynamics_criterion(x_t, x_t_pred)
+        loss_ae2 = self.dynamics_criterion(x_tau, x_tau_pred_dyn)
+        loss_dyn = self.dynamics_criterion(z_tau_pred, z_tau)
         loss_total = loss_ae1 * weight[0] + loss_ae2 * weight[1] + loss_dyn * weight[2]
         return loss_ae1, loss_ae2, loss_dyn, loss_total
 
-    def labels_losses(self, encodings, labels, weight):
-        # https://stackoverflow.com/questions/57428524/how-to-create-anchor-positive-and-anchor-negative-pairs-from-dataset-for-trainin 
-        raise NotImplementedError
-    
+    def labels_losses(self, encodings, triplets, weight):
+        contrastive_loss = self.labels_criterion(encodings[triplets['anchors']], encodings[triplets['positives']], encodings[triplets['negatives']])
+        return contrastive_loss * weight[3]
```

Comment:
Can just pass `weight[3]` as the argument so this is just `contrastive_loss * weight`.